internal/handlers/pg/msg_insert.go

```diff
@@ -118,8 +118,17 @@ func insertDocument(ctx context.Context, dbPool *pgdb.Pool, qp *pgdb.QueryParams
 		)
 	}
 
+	toInsert := d
+
+	if !toInsert.Has("_id") {
+		// We have to make a copy to avoid modifying the original document.
```

Comment:
Let's maybe also leave a comment to explain why we can't modify the original document?

Reply:
Updated comment.